src/components/Issues.js

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react'
+import { string } from 'prop-types'
+
+import Button from './Button'
+import { searchIssues } from '../utils/github'
+import Issue from './Issue'
+
+class Issues extends Component {
+  static propTypes = {
+    query: string,
+  }
+
+  static defaultProps = {
+    query: '',
+  }
+
+  state = {
+    issues: [],
+    hasNextPage: false,
+    endCursor: null,
+  }
+
+  componentDidMount() {
+    this.applySearch(this.props.query)
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.query !== this.props.query) {
+      const delay = 500
+
+      clearTimeout(this.setTimeoutId)
+
+      this.setTimeoutId = setTimeout(() => {
```

Comment:
I know you copied this code from me, but I'm wondering if there's a better name for `setTimeoutId`.